tests/e2e/ansible/install_docker.yml

```diff
@@ -68,4 +68,4 @@
         name: docker
         enabled: yes
         state: started
-  when: docker_exist.rc != 0 and ansible_distribution == "CentOS"
+  when: docker_exist.rc != 0 and ansible_distribution_file_variety == "RedHat"
```

Comment:
On CentOS Stream 9, `ansible_distribution_file_variety` is  `CentOS`. I suggest to stick with `ansible_distribution` like:
```
-  when: docker_exist.rc != 0 and ansible_distribution_file_variety == "RedHat"
+  when: docker_exist.rc != 0 and ansible_distribution in ["CentOS", "RedHat"]
```

The link https://download.docker.com/linux/centos/docker-ce.repo should also be changed to support fedora, rhel and centos.